aurora/model/aurora.py

```diff
@@ -316,8 +316,42 @@ def load_checkpoint_local(self, path: str, strict: bool = True) -> None:
                 d[f"decoder.atmos_heads.{name}.weight"] = weight[:, i]
                 d[f"decoder.atmos_heads.{name}.bias"] = bias[:, i]
 
+        #check if history size is compatible and adjust weights if necessary
+        if self.max_history_size > d["encoder.surf_token_embeds.weights.2t"].shape[2]:
+            d = self.adapt_checkpoint_max_history_size(d)
+        elif self.max_history_size < d["encoder.surf_token_embeds.weights.2t"].shape[2]:
+            assert False, f"Cannot load checkpoint with max_history_size {d['encoder.surf_token_embeds.weights.2t'].shape[2]} \
+                into model with max_history_size {self.max_history_size}" 
+                
         self.load_state_dict(d, strict=strict)
 
+    def adapt_checkpoint_max_history_size(self, checkpoint) -> Any:
+        """Adapt a checkpoint with smaller max_history_size to a model with a larger max_history_size
+        than the current model.
+        
+        If a checkpoint was trained with a larger max_history_size than the current model, this function
+        will assert fail to prevent loading the checkpoint. This is to prevent loading a checkpoint
+        which will likely cause the checkpoint to degrade is performance. 
+        
+        This implementation copies weights from the checkpoint to the model, duplicating the weights. 
+        """
+        #find all weights with prefix "encoder.surf_token_embeds.weights."
+        for name, weight in list(checkpoint.items()):
+            if name.startswith("encoder.surf_token_embeds.weights.") or name.startswith("encoder.atmos_token_embeds.weights."):
+                #this shouldn't get called with current logic but leaving here for future proofing and in cases where its called
+                #outside current context
+                assert weight.shape[2] <= self.max_history_size, f"Cannot load checkpoint with max_history_size {weight.shape[2]} \
+                    into model with max_history_size {self.max_history_size} for weight {name}"   
+            
+                # Initialize the new weight tensor with zeros
+                new_weight = torch.zeros((weight.shape[0], 1, self.max_history_size, weight.shape[3], weight.shape[4]))
+                # Copy the existing weights to the new tensor my duplicating the histories provided in to any new history dimensions
+                for j in range(new_weight.shape[2]):
+                    new_weight[:, :, j, :, :] = weight[:, :, j % weight.shape[2], :, :]
```

Comment:
I'm wondering about the `j % weight.shape[2]` logic. If you add additional history dimensions, perhaps it makes sense to set those weights to zero, so that the model doesn't use them at initialisation? In some way that makes sense, because that is how the model was trained before. Are you worried that this initialisation might make it hard for the model to learn to use the new history dimensions?

Reply:
In my initial adhoc testing I was under the impression that copying the previous weights was better than just 0s but based on your comment I decided to test if further.  Using 30 seperate batches from my validation set (default model vars and tuning on the small model/no fine tuning) the mse loss was averaged across all vars was about 20% better using 0s to fill.  So I've gone ahead and updated it filling with zeros.  Good catch!

Reply:
Thanks for the thorough investigation. That's really helpful!